ci/windows_dev_env/windows_deps_headless_installer.py

```diff
@@ -0,0 +1,365 @@
+
+
+def install_cuda():
+    # CUDA 9.2 and patches
+    logging.info("Installing CUDA 9.2 and Patches...")
+    cuda_9_2_file_path = download('https://developer.nvidia.com/compute/cuda/9.2/Prod2/network_installers2/cuda_9.2.148_win10_network')
+    run_command("PowerShell Rename-Item -Path {} -NewName \"{}.exe\"".format(cuda_9_2_file_path, cuda_9_2_file_path.split('\\')[-1]), shell=True)
+    cuda_9_2_file_path = cuda_9_2_file_path + '.exe'
+    run_command(cuda_9_2_file_path \
+    + ' -s nvcc_9.2' \
+    + ' cuobjdump_9.2' \
+    + ' nvprune_9.2' \
+    + ' cupti_9.2' \
+    + ' gpu_library_advisor_9.2' \
+    + ' memcheck_9.2' \
+    + ' nvdisasm_9.2' \
+    + ' nvprof_9.2' \
+    + ' visual_profiler_9.2' \
+    + ' visual_studio_integration_9.2' \
+    + ' demo_suite_9.2' \
+    + ' documentation_9.2' \
+    + ' cublas_9.2' \
+    + ' cublas_dev_9.2' \
+    + ' cudart_9.2' \
+    + ' cufft_9.2' \
+    + ' cufft_dev_9.2' \
+    + ' curand_9.2' \
+    + ' curand_dev_9.2' \
+    + ' cusolver_9.2' \
+    + ' cusolver_dev_9.2' \
+    + ' cusparse_9.2' \
+    + ' cusparse_dev_9.2' \
+    + ' nvgraph_9.2' \
+    + ' nvgraph_dev_9.2' \
+    + ' npp_9.2' \
+    + ' npp_dev_9.2' \
+    + ' nvrtc_9.2' \
+    + ' nvrtc_dev_9.2' \
+    + ' nvml_dev_9.2' \
+    + ' occupancy_calculator_9.2'
+    )
+    # Download patches and assume less than 100 patches exist
+    for patch_number in range(1, 100):
+        if patch_number == 100:
+            raise Exception('Probable patch loop: CUDA patch downloader is downloading at least 100 patches!')
+        cuda_9_2_patch_file_path = download("https://developer.nvidia.com/compute/cuda/9.2/Prod2/patches/{0}/cuda_9.2.148.{0}_windows".format(patch_number))
+        if cuda_9_2_patch_file_path == 404:
+            break
+        run_command("PowerShell Rename-Item -Path {} -NewName \"{}.exe\"".format(cuda_9_2_patch_file_path, cuda_9_2_patch_file_path.split('\\')[-1]), shell=True)
+        cuda_9_2_patch_file_path = cuda_9_2_patch_file_path + '.exe'
+        run_command("{} -s".format(cuda_9_2_patch_file_path))
+
+
+def add_paths():
+    # TODO: Add python paths (python -> C:\\Python37\\python.exe, python2 -> C:\\Python27\\python.exe)
+    logging.info("Adding Windows Kits to PATH...")
+    current_path = run_command("PowerShell (Get-Itemproperty -path 'hklm:\\system\\currentcontrolset\\control\\session manager\\environment' -Name Path).Path")
+    logging.debug("current_path: {}".format(current_path))
+    new_path = current_path + ";C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.16299.0\\x86;C:\\Program Files\\OpenBLAS-windows-v0_2_19\\bin"
+    logging.debug("new_path: {}".format(new_path))
+    run_command("PowerShell Set-ItemProperty -path 'hklm:\\system\\currentcontrolset\\control\\session manager\\environment' -Name Path -Value '" + new_path + "'")
+
+
+def has_gpu():
+    hwinfo = check_output(['powershell','gwmi', 'win32_pnpEntity'])
+    m = re.search('3D Video', hwinfo.decode())
```

Comment:
Is this too simple?check "NVIDIA"  at the same time?

Reply:
yes, this is crap I will remove it. but G3 doesn't show any "NVIDIA" I guess they didn't include much hw info.